packages/dev-server-core/src/utils.ts

```diff
@@ -3,10 +3,31 @@ import isStream from 'is-stream';
 import getStream from 'get-stream';
 import { isBinaryFile } from 'isbinaryfile';
 import path from 'path';
+import { parse as parseHtml } from 'parse5';
+import { query, predicates } from './dom5';
 
 const REGEXP_TO_BROWSER_PATH = new RegExp(path.sep === '\\' ? '\\\\' : path.sep, 'g');
 const REGEXP_TO_FILE_PATH = new RegExp('/', 'g');
 
+/**
+ * Appends a HTML fragment to the document of a given context and returns
+ * the transformed document.
+ */
+export function appendHtmlToDocument(context: Context, script: string): string | undefined {
```

Comment:
Let's make this function accept the HTML string, that will make it more reusable. Also `script` could rather be something like `appendedHtml` since it could be anything.